### PR TITLE
Repro code improvements and optimization.

### PR DESCRIPTION
**Pull request by krystian8207**

### CODE REVIEW

File: R/source_tblist.R [hunk 1]
```diff
@@ -614,7 +613,7 @@ cb_filter.multi_discrete.tblist <- function(
     },
     plot_data = function(data_object) {
       if (nrow(data_object[[dataset]])) {
-        data_object[[dataset]][names(values)] %>%
+        data_object[[dataset]][names(variables)] %>%
```
Comment (galachad) [inline R/source_tblist.R:616]: This was the plot error that was fixed here?

**Review by galachad — COMMENTED** [review]
LGTM for this changes.

The test are throwing the warning error for dplyr::filter.

```
Multi discrete filter works fine
data frame results in `filter()` are deprecated, use `if_any()` or `if_all()`.
```

I think we should fix deprecated dplyr warning in next release.

File: R/source_tblist.R (hunk 1 above)
Comment (krystian8207) [reply R/source_tblist.R:616]: yes it is

**Comment by krystian8207** [comment]
@galachad Let's merge now and I'll try to hide/fix the warning before the future release.
It's strongly related to https://github.com/tidyverse/dplyr/issues/6275.

**Pull request merged**